mat/eigen.go

```diff
@@ -212,6 +231,12 @@ func (e *Eigen) Factorize(a Matrix, left, right bool) (ok bool) {
 	return true
 }
 
+// Kind returns the SVDKind of the decomposition. If no decomposition has been
```

Comment:
Is this valid? What if the user called `Factorize(a, EigenNone)`?

Also, s/SVDKind/EigenKind/.

Reply:
Then Kind() also returns none?

We can change the behavior to return -1 if the factorization was unsuccessful (or not computed at all)?

Reply:
Fair enough. I read it as indication of success, but that was obviously just me.

Reply:
I'll switch it to return -1, so that way it can also be used as an indication of success.

Reply:
No, I think it's right as it is.